src/Engine/Engine_test.cc

```diff
@@ -24,8 +24,14 @@ OR OTHER DEALINGS IN THE SOFTWARE. */
 namespace rt {
```

Comment:
Can you provide more tests for this scene? A ray that fully intersect the object and a ray that have an antialiased pixel